backend/capellacollab/core/logging.py

```diff
@@ -0,0 +1,155 @@
+
+
+class LogExceptionMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next):
+        try:
+            return await call_next(request)
+        except Exception as exc:
+            logging.getLogger(request.url.path).exception(
+                msg=f'traceId={request.state.trace_id} message="{exc}"',
+                exc_info=True,
+            )
+            raise exc
+
+
+class LogRequestsMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next):
+        get_logger(request).info("request started")
+        response: Response = await call_next(request)
+        get_response_logger(request).info(
+            "request finished", status_code=response.status_code
+        )
+
+        return response
+
+
+class HealthcheckFilter(logging.Filter):
+    def filter(self, record: logging.LogRecord) -> bool:
+        return record.getMessage().find("/healthcheck") == -1
+
+
+class ReqLogAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        return (
+            "trace_id={trace_id} method={method} path={path} user={user} client={client} {msg_pref}{msg}".format(
+                trace_id=self.extra["trace_id"],
+                method=self.extra["method"],
+                path=self.extra["path"],
+                user=self.extra["user"],
+                client=self.extra["client"],
+                msg_pref="" if self.extra["chaining"] else 'message="',
+                msg=msg + ("" if self.extra["chaining"] else '"'),
+            ),
+            kwargs,
+        )
+
+
+class ErrorLogAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        error_code = kwargs.pop("error_code", self.extra["error_code"])
+        return (f'error_code={error_code} message="{msg}"', kwargs)
+
+
+class ResLogAdapter(logging.LoggerAdapter):
+    def process(self, msg, kwargs):
+        status_code = kwargs.pop("status_code", self.extra["status_code"])
+        return (f'status_code={status_code} message="{msg}"', kwargs)
+
+
+def get_general_logger(name: str, log_leveL=LOGGING_LEVEL) -> logging.Logger:
+    logger: logging.Logger = logging.getLogger(name)
+    logger.addFilter(HealthcheckFilter())
+    logger.setLevel(log_leveL)
+
+    if not logger.handlers:
+        handler = logging.StreamHandler()
+        formatter = logging.Formatter(
+            "time=%(asctime)s level=%(levelname)s function=%(funcName)s %(message)s"
+        )
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+
+    return logger
+
+
+def get_general_log_args(
+    request: Request, chaining=False
+) -> t.Dict[str, t.Any]:
+    client = request.client
+    return {
+        "trace_id": request.state.trace_id,
+        "method": request.method,
+        "path": request.url.path,
+        "user": request.state.user_name,
+        "client": client[0] + ":" + str(client[1]) if client else "unknown",
+        "chaining": chaining,
```

Comment:
`chaining` was used here to reduce/remove duplicate code. It essentially determined if `message={msg}` had to be included or if `msg` contained other key-value pairs, with the actual log message at the very end. However, as we discussed and you pushed, the new logging approach, where you can basically specify arbitrary key-value pairs within the `extra` dict, greatly simplifies logging so that `chaining` is no longer needed (since we now only have a single logger).